### PR TITLE
Enabled client endpoints for CF

### PR DESCRIPTION
**Pull request by eldadfux**
## What does this PR do?

Allow cloud functions to be executed from the client side by authorized users.

## Test Plan

Added new tests to simulate client execution of cloud funtions.

## Related PRs and Issues

#307 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes.

### CODE REVIEW

**Comment by christyjacob4** [comment]
@eldadfux by client, do you mean the server SDK ?

File: app/workers/functions.php [hunk 1]
```diff
@@ -507,7 +505,7 @@ public function cleanup(): void
                 $stdout = '';
                 $stderr = '';
 
-                if(Console::execute("docker stop {$first['name']}", '', $stdout, $stderr, 30) !== 0) {
+                if(Console::execute("docker rm -f {$first['name']}", '', $stdout, $stderr, 30) !== 0) {
```
Comment (christyjacob4) [inline app/workers/functions.php:508]: @eldadfux  we should probably stop the container before we remove it else `docker rm -f` would throw an error.

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Looks good 👍

File: app/workers/functions.php (hunk 1 above)
Comment (eldadfux) [reply app/workers/functions.php:508]: The -f option allow us to remove containers without needing to stop them. Its just work faster than sending 2 CLI commands.

**Comment by eldadfux** [comment]
> @eldadfux by client, do you mean the server SDK ?

More like opening cloud function API endpoints to be available from client side apps, mainly for executing pre-configured functions.

**Comment by christyjacob4** [comment]
If it is a client side app, Im confused by the usage of an API key in the request.
https://github.com/appwrite/appwrite/pull/788/files#diff-4a6c9e676aa23dca41808a3da457256a06c81ab23512ad7ced740696fd11541cR53

**Comment by eldadfux** [comment]
> If it is a client side app, Im confused by the usage of an API key in the request.
> https://github.com/appwrite/appwrite/pull/788/files#diff-4a6c9e676aa23dca41808a3da457256a06c81ab23512ad7ced740696fd11541cR53

Only the execution routes are available to the client (/v1/functions/[id]/executions). I have to create a function using the server API before I can execute it on the client.

**Comment by christyjacob4** [comment]
Okay makes sense now 👍  Thank you

File: app/workers/functions.php (hunk 1 above)
Comment (christyjacob4) [reply app/workers/functions.php:508]: Okay awesome 👍

**Review by christyjacob4 — APPROVED** [review]

**Review by TorstenDittmann — COMMENTED** [review]
Was a big one, but didn't see anything wrong 👍 

Great job!

Edit: Well, didn't see it was already merged 😆

**Pull request merged**